src/server/routes/items.js

```diff
@@ -150,5 +148,36 @@ router.get('/search/:searchText', async (req, res) => {
     })
   }
 })
+/**
+ * GET /items/filter/:filters
+ *
+ * Filtering items with selected inputs.
+ *
+ * @param filters: the JSON object that contains the filters
+ * @returns items
+ */
+router.get('/filter/:filters', async (req, res) => {
+  const rate = JSON.parse(req.params.filters).rating
+  const category = JSON.parse(req.params.filters).type
+  console.log(rate)
+  console.log(category)
+  if (req.params.filters !== undefined && rate !== undefined && category !== undefined) {
+    let items
+    if (rate === 'Unrated') {
+      if (category === 'Select item type...') {
+        items = await Item.find().populate('owner')
+      } else {
+        items = await Item.find({ type: category }).populate('owner')
+      }
+    } else if (category === 'Select item type...') {
+      items = await Item.find({ rating: rate }).populate('owner')
+    } else {
+      items = await Item.find({ rating: rate, type: category }).populate('owner')
+    }
+    res.send({
+      result: items
+    })
+  }
+})
```

Comment:
this logic could be better. don't rely on the client sending the default values of 'Select item type...' etc. those values could change. 

instead check for null or undefined, and have the client send null or undefined in the filter object if they haven't picked a value